nomenclature/processor/data_validator.py

```diff
@@ -86,9 +86,49 @@ def criteria(self):
         )
 
 
+class DataValidationCriteriaRange(DataValidationCriteria):
+    range: list[float] = Field(..., min_length=2, max_length=2)
+
+    @model_validator(mode="after")
+    def check_range_is_valid(self):
+        if self.range[0] > self.range[1]:
+            raise ValueError("Validation range is invalid: " + str(self.criteria))
+        return self
+
+    @computed_field
```

Comment:
Nice use of the `computed_field` decorator :+1:  To make things more explicit and follow the pydantic docs (https://docs.pydantic.dev/2.10/concepts/fields/#the-computed_field-decorator), I'd adding the property decorator as well for this and `lower_bound`.